### PR TITLE
Only set email sender if sender is not set

### PR DESCRIPTION
**Pull request by karenc**
Get the email default sender from `SiteSetting` and set it as the email
sender in `Email.__init__`.

Change app config `MAIL_DEFAULT_SENDER` to a property so we can guess
the email domain name.

Use `current_app.config['MAIL_DEFAULT_SENDER']` as the default for the
email default sender `SiteSetting` and remove
`app.utils.site_email_hostname`.

### CODE REVIEW

File: tests/modules/emails/test_email.py [hunk 1]
```diff
@@ -79,3 +81,16 @@ def test_email_templates():
...
             print(f'Email Template: {path}')
             raise
+
+
+def test_sender():
+    msg = Email(recipients=[test_recipient])
+    assert msg.sender == 'Codex Mailbot <changeme@example.com>'
+
+    SiteSetting.set_key_value('email_default_sender_name', 'Site Admin')
+    SiteSetting.set_key_value('email_default_sender_email', 'admin@codex.org')
```
Comment (naknomum) [inline tests/modules/emails/test_email.py:91]: `codex.org` is a real domain -- maybe use a subdomain of `example.org`, e.g. `codex.example.org`.  just to make sure we dont accidentally bother a real domain.

Comment (karenc) [reply tests/modules/emails/test_email.py:91]: Ah of course, I'll change it

Comment (naknomum) [reply tests/modules/emails/test_email.py:91]: btw, [RFC2606](https://datatracker.ietf.org/doc/html/rfc2606#section-3) specifically sets aside these as usable for testing, so no mailer will do anything sketchy while using these.  in theory. haha

**Comment by karenc** [comment]
@naknomum I changed the `MAIL_DEFAULT_SENDER` so now it's based on the `SERVER_NAME`:

```diff
diff --git a/config/base.py b/config/base.py
index 93bd6863..df477463 100644
--- a/config/base.py
+++ b/config/base.py
@@ -207,10 +207,16 @@ class EmailConfig(object):
     MAIL_USE_SSL = bool(os.getenv('MAIL_USE_SSL', False))
     MAIL_USERNAME = os.getenv('MAIL_USERNAME', 'dev@wildme.org')
     MAIL_PASSWORD = os.getenv('MAIL_PASSWORD', 'XXX')
-    MAIL_DEFAULT_SENDER = (
-        os.getenv('MAIL_DEFAULT_SENDER_NAME', 'Codex Mailbot'),
-        os.getenv('MAIL_DEFAULT_SENDER_EMAIL', 'changeme@example.com'),
-    )                                     
+                                          
+    @property                             
+    def MAIL_DEFAULT_SENDER(self):        
+        return (                          
+            os.getenv('MAIL_DEFAULT_SENDER_NAME', 'Do Not Reply'),
+            os.getenv(                    
+                'MAIL_DEFAULT_SENDER_EMAIL',
+                f'do-not-reply@{self.SERVER_NAME.replace("www.", "", 1)}',
+            ),                            
+        )                                 
                                           
                                           
 class ReCaptchaConfig(object):            
```

**Review by naknomum — APPROVED** [review]
looks good

**Comment by karenc** [comment]
The test doesn't work correctly until #544 is merged

**Pull request merged**